build/gulp/tasks/test-projects.tsx

```diff
@@ -0,0 +1,135 @@
+import * as fs from 'fs'
+import { task, series } from 'gulp'
+import * as rimraf from 'rimraf'
+import sh from '../sh'
+import * as mkdirp from 'mkdirp'
+import * as parseJson from 'parse-json'
+
+import config from '../../../config'
+import { tmpdir } from 'os'
+
+import * as path from 'path'
+import * as tmp from 'tmp'
+
+const pkg = require('../../../package.json')
+
+const { paths } = config
+
+const projectsPath = (...args) => paths.test('projects', ...args)
+const tsAppPath = (...args) => projectsPath('cra-ts', ...args)
+
+// ----------------------------------------
+// Clean
+// ----------------------------------------
+
+task('clean:test:projects:cra-ts', cb => {
+  rimraf(tsAppPath(), cb)
+})
+
+// ----------------------------------------
+// Build
+// ----------------------------------------
+
+// TS Project:
+//  - Create a new react test app
+//  - Update the App.tsx to include some stardust imports
+//  - Link our package
+//  - Try and run a build
+task('build:test:projects:cra-ts', async () => {
+  const appTSX = `import {
+  Avatar,
+  Button,
+  Header,
+  Image,
+  Input,
+  Popup,
+  Provider,
+  themes
+} from '@stardust-ui/react';
+import * as React from 'react';
+
+class App extends React.Component {
+  public render() {
+    return (
+      <Provider theme={themes.teams}>
+        <div>
+          <Popup trigger={<Button content="Popup" />} content="Popup content" />
+          <Avatar src="//placehold.it" />
+          <Button content="Click me" />
+          <Header content="This is " />
+          <Image src="//placehold.it" />
+          <Input placeholder="Type here" />
+        </div>
+      </Provider>
+    );
+  }
+}
+
+export default App;
+`
+
+  const log = msg => {
+    console.log()
+    console.log('='.repeat(80))
+    console.log('CRA TS:', msg)
+    console.log('='.repeat(80))
+  }
+
+  log('STEP 0. Preparing Stardust package..')
+  await sh('yarn build:dist')
+
+  const stardustPackageFilename = (await sh(`npm pack`, true)).trim()
+  log(`Stardust package is published: ${paths.base(stardustPackageFilename)}`)
+
+  ////////
+
+  try {
+    log('STEP 1. Create test React project with TSX scripts..')
+    const testAppDir = path.resolve(tmp.dirSync().name, 'test')
+    const runInTestApp = cmd => sh(`cd ${testAppDir} && ${cmd}`)
+
+    fs.mkdirSync(testAppDir)
+
+    await runInTestApp(`yarn create react-app . --scripts-version=react-scripts-ts`)
```

Comment:
Heads up, this appears to install a global on the users machine:
https://yarnpkg.com/lang/en/docs/cli/create/